Lib/fontbakery/profiles/path.py

```diff
@@ -0,0 +1,225 @@
+                    if line == "x-height" and (
+                        len(glyphname) > 1 or glyphname[0].isupper()
+                    ):
+                        continue
+                    if close_but_not_on(yExpected, node.y, ALIGNMENT_MISS_EPSILON):
+                        warnings.append(
+                            f"{glyphname}: X={node.x},Y={node.y} (should be at {line} {yExpected}?)"
+                        )
+
+    if warnings:
+        formatted_list = "\t* " + pretty_print_list(warnings, sep="\n\t* ")
+        yield WARN, Message(
+            "found-misalignments",
+            f"The following glyphs have on-curve points which"
+            f" have potentially incorrect y coordinates:\n"
+            f"{formatted_list}",
+        )
+    else:
+        yield PASS, ("Y-coordinates of points fell on appropriate boundaries.")
+
+
+@check(
+    id="com.google.fonts/check/path_short_segments",
+    conditions=["paths_dict", "is_not_variable_font"],
+)
+def com_google_fonts_check_path_short_segments(ttFont, paths_dict):
+    """Are any segments inordinately short?"""
+    warnings = []
+    for glyphname, paths in paths_dict.items():
+        for p in paths:
+            path_length = p.length
+            segments = p.asSegments()
+            if not segments:
+                continue
+            prev_was_line = len(segments[-1]) == 2
+            for seg in p.asSegments():
+                if seg.length < SHORT_PATH_EPSILON * path_length and (
+                    prev_was_line or len(seg) > 2
+                ):
+                    warnings.append(f"{glyphname} contains a short segment {seg}")
+                prev_was_line = len(seg) == 2
+    if warnings:
+        formatted_list = "\t* " + pretty_print_list(warnings, sep="\n\t* ")
+        yield WARN, Message(
+            "found-short-segments",
+            f"The following glyphs have segments which seem very short:\n"
+            f"{formatted_list}",
+        )
+    else:
+        yield PASS, ("No short segments were found.")
+
+
+@check(
+    id="com.google.fonts/check/path_colinear_vectors",
+    conditions=["paths_dict", "is_not_variable_font"],
+)
+def com_google_fonts_check_path_colinear_vectors(ttFont, paths_dict):
+    """Do any segments have colinear vectors?"""
+    warnings = []
+    for glyphname, paths in paths_dict.items():
+        for p in paths:
+            segments = p.asSegments()
+            if not segments:
+                continue
+            for i in range(0, len(segments)):
+                prev = segments[i - 1]
+                this = segments[i]
+                if len(prev) == 2 and len(this) == 2:
+                    if (
+                        abs(prev.tangentAtTime(0).angle - this.tangentAtTime(0).angle)
+                        < COLINEAR_EPISON
+                    ):
+                        warnings.append(glyphname)
+
+    if warnings:
+        formatted_list = "\t* " + pretty_print_list(list(set(warnings)), sep="\n\t* ")
```

Comment:
here you convert a list into a set and back into a list again. Is that meant to remove duplicates?

Also, it would be good to sort this list to avoid problems like the one described at #3038